### PR TITLE
Bump rust version up to 1.43.0

### PR DESCRIPTION
**Pull request by aleksuss**
## Overview

---

See: https://jira.bf.local/browse/ECR-XYZW

### Definition of Done

- [ ] There are no TODOs left in the merged code
- [ ] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [ ] The [coding guidelines] are followed
- [ ] Public API has proper documentation
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: components/crypto/src/lib.rs [hunk 1]
```diff
@@ -550,7 +550,11 @@ fn verify_keys_match(public_key: &PublicKey, secret_key: &SecretKey) -> bool {
 
 #[cfg(test)]
 mod tests {
-    use super::*;
```
Comment (slowli) [inline components/crypto/src/lib.rs:553]: The relevant [clippy docs](https://rust-lang.github.io/rust-clippy/master/index.html#wildcard_imports) state:

> `use super::*` is allowed in test modules. This is defined as any module with “test” in the name.

Is this not the case? IMO, `use super::*` for test modules does make sense.

File: components/keys/examples/key_generator.rs [hunk 2]
```diff
@@ -13,7 +13,6 @@
 // limitations under the License.
 
 use exonum_keys::*;
```
Comment (slowli) [inline components/keys/examples/key_generator.rs:15]: Why is wildcard import not affected? 🙃 Intuitively, it looks more "suspicious" than ones in the test modules, for example.

File: exonum-node/src/events/internal.rs [hunk 3]
```diff
@@ -93,9 +93,12 @@ mod tests {
         helpers::Height,
         messages::Verified,
     };
+    use futures::{SinkExt, StreamExt};
     use pretty_assertions::assert_eq;
 
-    use super::*;
+    use super::{
+        mpsc, BinaryValue, InternalEvent, InternalPart, InternalRequest, Message, SignedMessage,
```
Comment (slowli) [inline exonum-node/src/events/internal.rs:100]: Nit: Why are `mpsc` / `BinaryValue` re-imported from `super`, whereas `SinkExt` / `StreamExt` are imported directly from their definition location (`futures`)?

File: components/crypto/src/lib.rs (hunk 1 above)
Comment (aleksuss) [reply components/crypto/src/lib.rs:553]: I followed by clippy suggestions only 🙃

File: components/keys/examples/key_generator.rs (hunk 2 above)
Comment (aleksuss) [reply components/keys/examples/key_generator.rs:15]: Weird. But clippy skipped it.

**Pull request merged**